source/samples/pubsub/PubSubFeature.cpp

```diff
@@ -40,8 +99,28 @@ int PubSubFeature::init(
     thingName = *config.thingName;
     pubTopic = config.pubSub.publishTopic.value();
     subTopic = config.pubSub.subscribeTopic.value();
-    pubFile = config.pubSub.publishFile.has_value() ? config.pubSub.publishFile.value() : "";
-    subFile = config.pubSub.subscribeFile.has_value() ? config.pubSub.subscribeFile.value() : "";
+
+    if (config.pubSub.publishFile.has_value() && !config.pubSub.publishFile->empty())
+    {
+        pubFile = config.pubSub.publishFile->c_str();
+    }
+    pubFile = FileUtils::ExtractExpandedPath(pubFile);
+
+    if (!createPubSub(config, pubFile))
+    {
+        LOG_ERROR(TAG, "Failed to create publish directory or file");
```

Comment:
@xlcheng1 Failure to create the pub sub files PubSubFeature::init is not being propagated back to caller, I think that should be changed. Same applies below.  What do you think @HarshGandhi-AWS?

Reply:
In DC right now we are not considering failures in sample as Fatal error. We should consider changing this behavior.

Reply:
will address it in a separate PR